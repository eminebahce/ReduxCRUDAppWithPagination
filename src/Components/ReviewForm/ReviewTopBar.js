import React, {Component} from 'react';
import {AppBar, Toolbar, Typography, IconButton, Grid} from '@material-ui/core'
import Icon from '@material-ui/core/Icon'
import MenuIcon from '@material-ui/icons/Menu';
import {withStyles} from '@material-ui/core/styles';

const styles = {
    grow: {
        flexGrow: 1,
        marginLeft: 10,
        marginRight: 10
    },
    menuButton: {
        marginLeft: -12,
    }
};

class ReviewTopBar extends Component {
    render() {
        return (
            <div data-test='top-bar'>
                <Grid container justify="center" alignContent="center">
                    <Grid item xs={6}>
                        <AppBar color="primary" position="static" data-test='app-bar'>
                            <Toolbar>
                                <IconButton style={styles.menuButton} color="inherit" aria-label="Menu">
                                    <MenuIcon/>
                                </IconButton>
                                <Icon>star</Icon>
                                <span><Typography variant="h6" style={styles.grow}
                                                  color="inherit">Reviews</Typography></span>
                                <Icon>star</Icon>
                            </Toolbar>
                        </AppBar>
                    </Grid>
                </Grid>
            </div>
        );
    }
}

export default withStyles(styles)(ReviewTopBar);
