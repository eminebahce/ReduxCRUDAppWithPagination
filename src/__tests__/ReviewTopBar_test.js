import React from 'react';
import {shallow} from 'enzyme';
import ReviewTopBar from '../../src/Components/ReviewForm/ReviewTopBar';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import store from "../store";
import {Grid, AppBar} from '@material-ui/core';

describe('<ReviewTopBar /> component', () => {
   it('should render without errors', () => {
        const component = shallow(<ReviewTopBar />);
        expect(component.find(`[data-test='top-bar']`)).toBeTruthy();
    });
    it('should render without errors', () => {
        const component = shallow(<ReviewTopBar />);
        expect(component.find(`[data-test='app-bar']`)).toBeTruthy();
    });

    describe('renders ReviewTopBar element with AppBar', () => {
        it('should not return error', () => {
            const component = renderer.create(
                <Provider store={store}>
                    <div>
                        <Grid>
                        <AppBar />
                        </Grid>
                    </div>
                </Provider>
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

});