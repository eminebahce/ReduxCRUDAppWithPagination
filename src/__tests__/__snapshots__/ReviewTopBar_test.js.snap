// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewTopBar /> component renders ReviewTopBar element with AppBar should not return error 1`] = `
<div>
  <div
    className=""
  >
    <header
      className="MuiPaper-root-109 MuiPaper-elevation4-115 MuiAppBar-root-100 MuiAppBar-positionFixed-101 MuiAppBar-colorPrimary-107 mui-fixed"
    />
  </div>
</div>
`;
